Fix typecheck error in `members.update` squad-scope check (callerUserId nullability)

## What to build

`npm run typecheck` reports an error in the squad-scope check inside `members.update`'s body:

```
server/apis/members.server.ts(117,65): error TS2769: No overload matches this call.
  Argument of type 'string | null' is not assignable to parameter of type 'string | ObjectID | Selector<Member> | undefined'.
```

The offending call:

```ts
// server/apis/members.server.ts:117
const viewer = await MembersCollection.findOneAsync(callerUserId);
```

`callerUserId` is typed `string | null` from its enclosing scope. The runtime is safe because `runMutation`'s auth gate has already rejected null `userId` before the body executes (`members.update` does not opt into `allowsAnonymous`). The type system can't see this — the auth check is in a separate module from the body, and the type information doesn't flow across that seam.

## Investigation context

A few possible shapes for the fix; the right one depends on how invasive you want to be:

- **Local narrow** at the body's entry point: `if (!callerUserId) throw new Meteor.Error(401, 'Unauthorized');` — simple but duplicates what `runMutation` already enforces.
- **Pass `userId` through the runMutation context** so the body receives a guaranteed-string `userId`. Cleanest, but touches the wrapper signature; should be considered against the existing pattern in #97/CONTEXT.md.
- **Type assertion at the call site** (`callerUserId!`): minimal change, weakest signal, future-fragile.

The PR review (post-#114/#115) flagged "no observable change in passing test outcomes" as the success bar; preserving the squad-scope reject behavior is what matters.

## Acceptance criteria

- [ ] `npm run typecheck` passes on `main` with this specific TS2769 resolved
- [ ] All 182 existing tests stay green
- [ ] Squad-scope reject behavior in `members.update` is unchanged (verified by `tests/server/membersMethods.test.ts`)

## Blocked by

None - can start immediately

--- a/server/apis/members.server.ts
+++ b/server/apis/members.server.ts
@@ -114,7 +114,7 @@ if (Meteor.isServer) {
           const targetMember = await getMemberById(targetId);
           const role = await getUserRole(callerUserId);
           if (!isOfficerOrAdmin(role)) {
-            const viewer = await MembersCollection.findOneAsync(callerUserId);
+            const viewer = await MembersCollection.findOneAsync(callerUserId!);
             if (viewer?.profile?.squadId && targetMember?.profile?.squadId !== viewer.profile.squadId) {
               throw new Meteor.Error(403, 'Cannot update members outside your squad');
             }

